R/utilities-mpi-error-handling.R

```diff
@@ -0,0 +1,153 @@
+#' @title checkSimulationLoaded
+#' @description Check that the simulation has been loaded on the core.
+#' @param simulation the simulation object to be checked if loaded successfully
+#' @param simulationFilePath the path to the simulation object to be checked if loaded successfully
+#' @export
+checkSimulationLoaded <- function(simulation, simulationFilePath) {
```

Comment:
I don't understand this method. What does that mean loaded?
How can you have an instance of a simulation that is not loaded? It is not possible to construct a simulation otherwise

Reply:
From a code perspective, this can be simplified by returning 

```R
checkSimulationLoaded <- function(simulation, simulationFilePath) {
   identical(simulation$sourceFile, simulationFilePath)
}
```

or 
```R
checkSimulationLoaded <- function(simulation, simulationFilePath) {
  return( identical(simulation$sourceFile, simulationFilePath))
}
```

Reply:
I want to check that `sim` is loaded correctly on each core.  If I initialize `sim `to NULL on the core, then load the simulation to `sim `on the core, then I can either
1) check that `sim` is not null, or
2) perform the above check (ie that the simulation was loaded using the correct sourceFile

is 2 overkill?

Reply:
oh i see.  
Wouldn't it crash if simulation is null?